components/script/task_queue.rs

```diff
@@ -53,13 +53,18 @@ impl<T: QueuedTaskConversion> TaskQueue<T> {
 
     /// Process incoming tasks, immediately sending priority ones downstream,
     /// and categorizing potential throttles.
-    fn process_incoming_tasks(&self) {
-        let mut non_throttled: Vec<T> = self.port
-            .try_iter()
-            .filter(|msg| !msg.is_wake_up())
-            .collect();
+    fn process_incoming_tasks(&self, first_msg: T) {
+        let mut incoming = Vec::with_capacity(self.port.len() + 1);
+        if !first_msg.is_wake_up() {
+            incoming.push(first_msg);
+        }
+        while let Some(msg) = self.port.try_recv() {
+            if !msg.is_wake_up() {
+                incoming.push(msg);
+            }
+        }
```

Comment:
this is a new change, trying to reduce the number of vec created, and the `with_capacity` seems worthwhile...